File: Windows App/Services/ParameterValidator.cs
using System.Collections.Generic;
using System.Linq;
using Voxa.Models;

namespace Voxa.Services
{
    /// <summary>
    /// Central place for "is this value sane" checks, used before a batch run starts
    /// and before a preset is saved, so bad values are caught with a clear message
    /// instead of surfacing as a cryptic FFmpeg failure per file.
    /// </summary>
    public static class ParameterValidator
    {
        public const int MinSampleRate = 8000;
        public const int MaxSampleRate = 192000;
        public const double MinVolumeDb = -30;
        public const double MaxVolumeDb = 30;
        public const double MinSpeed = 0.25;
        public const double MaxSpeed = 4.0;
        public const int MinBitrate = 32;
        public const int MaxBitrate = 320;
        public const double MinSilencePaddingSec = 0;
        public const double MaxSilencePaddingSec = 30;

        public static List<string> Validate(ProcessingParameters p)
        {
            var errors = new List<string>();

            if (string.IsNullOrWhiteSpace(p.OutputFormat) ||
                !AudioFileFilter.SupportedOutputFormats.Contains(p.OutputFormat.ToLowerInvariant()))
            {
                errors.Add($"'{p.OutputFormat}' is not a supported output format.");
            }

            if (!p.KeepOriginalSampleRate && (p.SampleRateHz < MinSampleRate || p.SampleRateHz > MaxSampleRate))
            {
                errors.Add($"Sample rate must be between {MinSampleRate:N0} Hz and {MaxSampleRate:N0} Hz.");
            }

            if (p.VolumeChangeDb < MinVolumeDb || p.VolumeChangeDb > MaxVolumeDb)
            {
                errors.Add($"Volume change must be between {MinVolumeDb} dB and {MaxVolumeDb} dB.");
            }

            if (p.SpeedMultiplier < MinSpeed || p.SpeedMultiplier > MaxSpeed)
            {
                errors.Add($"Speed must be between {MinSpeed}x and {MaxSpeed}x.");
            }

            if (p.BitrateKbps < MinBitrate || p.BitrateKbps > MaxBitrate)
            {
                errors.Add($"Bitrate must be between {MinBitrate} kbps and {MaxBitrate} kbps.");
            }

            if (p.SilencePaddingStartSec < MinSilencePaddingSec || p.SilencePaddingStartSec > MaxSilencePaddingSec)
            {
                errors.Add($"Starting silence must be between {MinSilencePaddingSec} and {MaxSilencePaddingSec} seconds.");
            }

            if (p.SilencePaddingEndSec < MinSilencePaddingSec || p.SilencePaddingEndSec > MaxSilencePaddingSec)
            {
                errors.Add($"Ending silence must be between {MinSilencePaddingSec} and {MaxSilencePaddingSec} seconds.");
            }

            return errors;
        }
    }
}

File: Android App/Voxa.Android/app/src/main/java/com/voxa/app/core/ParameterValidator.kt
package com.voxa.app.core

import com.voxa.app.model.ProcessingParameters

object ParameterValidator {
    private const val minSampleRate = 8000
    private const val maxSampleRate = 192000
    private const val minVolumeDb = -30.0
    private const val maxVolumeDb = 30.0
    private const val minSpeed = 0.25
    private const val maxSpeed = 4.0
    private const val minBitrate = 32
    private const val maxBitrate = 320
    private const val minSilencePaddingSec = 0.0
    private const val maxSilencePaddingSec = 30.0

    fun validate(parameters: ProcessingParameters): List<String> = buildList {
        if (parameters.outputFormat.lowercase() !in AudioFileFilter.supportedOutputFormats) {
            add("'${parameters.outputFormat}' is not a supported output format.")
        }

        if (!parameters.keepOriginalSampleRate &&
            parameters.sampleRateHz !in minSampleRate..maxSampleRate
        ) {
            add("Sample rate must be between $minSampleRate Hz and $maxSampleRate Hz.")
        }

        if (parameters.volumeChangeDb !in minVolumeDb..maxVolumeDb) {
            add("Volume change must be between $minVolumeDb dB and $maxVolumeDb dB.")
        }

        if (parameters.speedMultiplier !in minSpeed..maxSpeed) {
            add("Speed must be between ${minSpeed}x and ${maxSpeed}x.")
        }

        if (parameters.bitrateKbps !in minBitrate..maxBitrate) {
            add("Bitrate must be between $minBitrate kbps and $maxBitrate kbps.")
        }

        if (parameters.silencePaddingStartSec !in minSilencePaddingSec..maxSilencePaddingSec) {
            add("Starting silence must be between $minSilencePaddingSec and $maxSilencePaddingSec seconds.")
        }

        if (parameters.silencePaddingEndSec !in minSilencePaddingSec..maxSilencePaddingSec) {
            add("Ending silence must be between $minSilencePaddingSec and $maxSilencePaddingSec seconds.")
        }
    }
}
